### PR TITLE
Use libp2p v0.41 and libp2p-core v0.30

### PR DESCRIPTION
**Pull request by Alex6323**
# Description of change

Bumps `libp2p` dep to `0.41.0` and `libp2p-core` dep to `0.30.0`.

## Links to any relevant issues

Addresses #1209 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Unit tests pass.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-network/bee-autopeering/Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ bytes = { version = "1.0", default-features = false }
 hash32 = { version = "0.2.1", default-features = false }
 hex = { version = "0.4.3", default-features = false }
 iota-crypto = { version = "0.9.1", default-features = false, features = [ "ed25519", "random", "sha" ] }
-libp2p-core = { version = "0.29.0", default-features = false }
+libp2p-core = { version = "0.30.0", default-features = false }
```
Comment (thibault-martinez) [inline bee-network/bee-autopeering/Cargo.toml:22]: Should the autopeering crate also be bumped then ?

Comment (Alex6323) [reply bee-network/bee-autopeering/Cargo.toml:22]: oh, of course

File: bee-network/bee-gossip/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "bee-gossip"
-version = "0.5.0"
+version = "0.5.1"
```
Comment (thibault-martinez) [inline bee-network/bee-gossip/Cargo.toml]: Needs to be a minor since your are bumping actual deps

Comment (thibault-martinez) [reply bee-network/bee-gossip/Cargo.toml]: Same for the others

File: bee-network/bee-autopeering/Cargo.toml (hunk 1 above)
Comment (Alex6323) [reply bee-network/bee-autopeering/Cargo.toml:22]: Done

File: bee-network/bee-gossip/Cargo.toml (hunk 2 above)
Comment (Alex6323) [reply bee-network/bee-gossip/Cargo.toml]: yea

File: bee-network/bee-gossip/src/swarm/protocols/iota_gossip/protocol.rs [hunk 3]
```diff
@@ -116,7 +118,14 @@ impl NetworkBehaviour for IotaGossipProtocol {
     /// **libp2p docs**:
     ///
     /// Informs the behaviour about a newly established connection to a peer.
-    fn inject_connection_established(&mut self, peer_id: &PeerId, conn_id: &ConnectionId, endpoint: &ConnectedPoint) {
+
```
Comment (grtlr) [inline bee-network/bee-gossip/src/swarm/protocols/iota_gossip/protocol.rs]:
```suggestion
```

File: bee-network/bee-gossip/src/swarm/protocols/iota_gossip/protocol.rs [hunk 4]
```diff
@@ -204,7 +213,14 @@ impl NetworkBehaviour for IotaGossipProtocol {
     /// A call to this method is always paired with an earlier call to
     /// `inject_connection_established` with the same peer ID, connection ID and
     /// endpoint.
-    fn inject_connection_closed(&mut self, peer_id: &PeerId, _: &ConnectionId, _: &ConnectedPoint) {
+
```
Comment (grtlr) [inline bee-network/bee-gossip/src/swarm/protocols/iota_gossip/protocol.rs]:
```suggestion
```

**Review by thibault-martinez — DISMISSED** [review]

**Review by grtlr — DISMISSED** [review]

**Review by Adam-Gleave — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**